raster/src/main/scala/geotrellis/raster/io/geotiff/Float32GeoTiffSegmentCollection.scala

```diff
@@ -16,22 +16,18 @@
 
 package geotrellis.raster.io.geotiff
 
-import geotrellis.raster._
-import geotrellis.raster.io.geotiff.compression._
-import spire.syntax.cfor._
-
 trait Float32GeoTiffSegmentCollection extends GeoTiffSegmentCollection {
   type T = Float32GeoTiffSegment
 
   val bandType = Float32BandType
-  val noDataValue: Option[Float]
+  def noDataValue: Option[Float]
 
-  lazy val createSegment: Int => Float32GeoTiffSegment = noDataValue match {
+  lazy val decompressGeoTiffSegment = noDataValue match {
     case None =>
-      { i: Int => new Float32RawGeoTiffSegment(getDecompressedBytes(i)) }
-    case Some(nd) if (nd == Float.NaN) =>
-      { i: Int => new Float32ConstantNoDataGeoTiffSegment(getDecompressedBytes(i)) }
+      (i: Int, bytes: Array[Byte]) => new Float32RawGeoTiffSegment(decompressor.decompress(bytes, i))
+    case Some(nd) if nd.isNaN =>
```

Comment:
We always use `java.lang.Float.isNaN(nd)`, because for some reason it's significantly faster than `.isNaN`